text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+## Current state of Cargo/ crates.io tooling
+
+At present, crates.io provides the following information about published crates:
+
+* checksum of crate contents
+* crate author/ownership via GitHub
+
+Our current reality is that:
+
+* The verification of the above information relies on a third party (crates.io) not being compromised.
+* In the event that crates.io is compromised, crates pulled from crates.io cannot be verified.
+* There is no way to establish the identity of the system that built the artifact.
+* There is only the limited identity metadata associated with the GH ID that is the source of the artifact.
+
+By executing on the adoption of Sigstore and structuring crates.io metadata around the ability to present information capable of driving policy based consumption (both on the desktop and in the build and runtime environments), the Rust community will take a large and important step in evolving the its tooling to meet the demands of a rapidly evolving software supply chain ecosystem and of the security and regulatory demands that are coming into play.
+
+One of the first principles for considering the implementing this type of tooling in the community ecosystem is making sure that the developer experience remains a Tier 1 consideration. Historically, when security specific capabilities are brought into the mix the impact to productivity and developer experience has been of such a level that very often the security “stuff” gets ignored or just falls to the wayside (see PGP signing as an example). This is one area where Sigstore truly shines. From the outset Sigstore was designed to be a toolchain that would be a trivial impact to the day to day workflow of a developer as to be a non-issue. The cosign application is easy to use on the desktop (and fairly easy to drop into a build system)  and the integration with many widely used IdP implementations (GH, GCP, etc) make the workflow simple. There are no long-term keys to manage, nor any other infrastructure when you are utilizing the public service. Further the verification service (Rekor) is accessed through a well documented API making both desktop CLI and machine level access a straight-forward proposition.
+
+While not a part of this RFC, end users can implement additional projects such as [in-toto](https://in-toto.io/) to build more advanced verification and attestation of software.
+
+## Goals
+
+This proposal will enhance cargo and crates.io through adoption of the Sigstore capabilities and workflows for supporting both use cases as outlined above. Specifically, 
+
+* Adopt Sigstore infrastructure to allow for the signing of crates for use both above.
+* Facilitate workflows in both Cargo and crates.io to allow for signature generation in the former case and verification in the latter case of crates and their dependencies by crate consumers (a note on dependencies: the depth level should be 1 for verification in this first phase)
```

Comment:
This sentence is a train wreck. It asks for:

* Signature generation for cargo
* Signature verification for crates.io
* 1-depth dependency signing (dependencies only, not dependencies of dependencies)